### ISSUE TITLE
ICE failed error happened on FF57.

### ISSUE THREAD

**Issue by KensakuKOMATSU** [meetecho/janus-gateway#1119]
Hi,

I found that ICE failed error happened on FF57.0.3, High Sierra and FF58.0b13, El Capitan (58 is still beta, though).

I used Echo and Streaming demo app with v0.2.6 and current master branch (default configuration).

WARN and ERR log was show like below.

```
[WARN] [6952112567877392] ICE failed for component 1 in stream 1, but let's give it some time... (trickle pending, answer received, alert not set)
[WARN] [6952112567877392] ICE failed for component 1 in stream 1, but let's give it some time... (trickle received, answer received, alert not set)
[ERR] [ice.c:janus_ice_check_failed:1627] [6952112567877392] ICE failed for component 1 in stream 1...
```

**Comment by lminiero** [comment]
I'm not aware of any bugs, but I don't have any Mac so not sure. Check the Admin API to figure out why I D fails: http://www.meetecho.com/blog/understanding-the-janus-admin-api/

**Comment by lminiero** [comment]
Any update?

**Comment by KensakuKOMATSU** [comment]
Sorry for late reply, since I had been new years holiday.

Thanks for your comment. I'll check this issue with Admin API.

**Comment by KensakuKOMATSU** [comment]
One thing I have to notice is that my environment is bit specific. Janus is running inside Vagrant and browser is on top of Host computer. That means only Janus is under NAT.

I tested several patterns on this env, with stun and/or turn settings, but I found that with every patterns FF failed.

I posted admin API logs (it's really convenient API, thanks!!) in gist.

https://gist.github.com/KensakuKOMATSU/71cf5d83db9a467e0111f93a038866c1

I believe that if turn setting in ``janus.cfg`` is applied to sample web app, it works fine, though.
